manim/utils/cfgwriter.py

```diff
@@ -26,7 +25,7 @@ def check_valid_colour(color):
 def main():
     console = Console()
     config = configparser.ConfigParser()
-    config.read("manim.cfg")
+    config.read("../default.cfg")
```

Comment:
I think this doesn't work. Do check it. Because it reads for the file `manim.cfg` in cwd and creates one if it doesn't exists. So like what you are reading would be wrong.

Reply:
> Because it reads for the file manim.cfg in cwd and creates one if it doesn't exists

No. Your script will create a `manim.cfg` file in the CWD anyway, even if another manim.cfg file is in the cwd:

https://github.com/naveen521kk/manim/blob/5f263cba49fd6c913084dde6d679e424506eb3b8/manim/utils/cfgwriter.py#L65-L66

And yes, it does create the proper files in the proper places. The files created before and after these changes were identical.

However, it appears as though relative paths need to be made absolute, and that is why it wasn't reading it properly. The newest commit will fix this.

Also, reading from `default.cfg`, which already has defaults for all the other parameters, means that the the created `manim.cfg` file will have parameters from `default.cfg` as well, instead of _just_ rich oriented configs.

Reply:
Before I made this change, the contents of `manim.cfg` were:

```
[log.color]
logging.keyword = bold yellow
logging.level.notset = dim
logging.level.debug = green
logging.level.info = blue
logging.level.warning = red
logging.level.error = red bold
logging.level.critical = red bold reverse
log.level = 
log.time = cyan dim
log.message = 
log.path = dim
```

After making these changes, manim.cfg now includes not only the above lines, but also the parameters from default.cfg, so it's better for use as a general purpose config writer.

```
[CLI]
write_to_movie = True
save_last_frame = False
write_all = False
save_pngs = False
save_as_gif = False
preview = False
show_file_in_finder = False
quiet = False
sound = False
output_file = 
leave_progress_bars = False
background_color = BLACK
background_opacity = 1
from_animation_number = 0
upto_animation_number = -1
media_dir = ./media
png_mode = RGB
movie_file_extension = .mp4
frame_rate = 60
pixel_height = 1440
pixel_width = 2560

[transparent]
png_mode = RGBA
movie_file_extension = .mov
background_opacity = 0

[fourk_quality]
pixel_height = 2160
pixel_width = 3840
frame_rate = 60

[high_quality]
pixel_height = 1080
pixel_width = 1920
frame_rate = 60

[medium_quality]
pixel_height = 720
pixel_width = 1280
frame_rate = 30

[low_quality]
pixel_height = 480
pixel_width = 854
frame_rate = 15

[dry_run]
write_to_movie = False
write_all = False
save_last_frame = False
save_pngs = False
save_as_gif = False

[streaming]
live_stream_name = LiveStream
twitch_stream_key = YOUR_STREAM_KEY
streaming_protocol = tcp
streaming_ip = 127.0.0.1
streaming_port = 2000
streaming_client = ffplay
streaming_url = %(STREAMING_PROTOCOL)s://%(STREAMING_IP)s:%(STREAMING_PORT)s?listen
streaming_console_banner = Manim is now running in streaming mode.
	Stream animations by passing them to manim.play(), e.g.
	>>> c = Circle()
	>>> manim.play(ShowCreation(c))

[log.color]
logging.keyword = bold yellow
logging.level.notset = dim
logging.level.debug = green
logging.level.info = blue
logging.level.warning = red
logging.level.error = red bold
logging.level.critical = red bold reverse
log.level = 
log.time = cyan dim
log.message = 
log.path = dim
```